src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -94,24 +120,73 @@ void CommissionerDiscoveryController::OnUserDirectedCommissioningRequest(UDCClie
     mReady = false;
     Platform::CopyString(mCurrentInstance, state.GetInstanceName());
     mPendingConsent = true;
-    char rotatingDeviceIdHexBuffer[RotatingDeviceId::kHexMaxLength];
-    Encoding::BytesToUppercaseHexString(state.GetRotatingId(), state.GetRotatingIdLength(), rotatingDeviceIdHexBuffer,
-                                        RotatingDeviceId::kHexMaxLength);
+    char rotatingIdString[Dnssd::kMaxRotatingIdLen * 2 + 1];
+    Encoding::BytesToUppercaseHexString(state.GetRotatingId(), state.GetRotatingIdLength(), rotatingIdString,
+                                        sizeof(rotatingIdString));
 
     ChipLogDetail(Controller,
                   "------PROMPT USER: %s is requesting permission to cast to this TV, approve? [" ChipLogFormatMEI
                   "," ChipLogFormatMEI ",%s,%s]",
                   state.GetDeviceName(), ChipLogValueMEI(state.GetVendorId()), ChipLogValueMEI(state.GetProductId()),
-                  state.GetInstanceName(), rotatingDeviceIdHexBuffer);
+                  state.GetInstanceName(), rotatingIdString);
     if (mUserPrompter != nullptr)
     {
         mUserPrompter->PromptForCommissionOKPermission(state.GetVendorId(), state.GetProductId(), state.GetDeviceName());
     }
     ChipLogDetail(Controller, "------Via Shell Enter: controller ux ok|cancel");
 }
 
+/// Callback for getting execution into the main chip thread
+void CallbackOk(System::Layer * aSystemLayer, void * aAppState)
+{
+    ChipLogError(AppServer, "UX Ok: now on main thread");
+    GetCommissionerDiscoveryController()->InternalOk();
```

Comment:
Shouldn't we just pass the controller via aAppState instead of relying on globals?

Reply:
fixed